src/commands/registry.rs

```diff
@@ -92,8 +112,27 @@ impl Pull {
     /// Pull a Spin application from an OCI registry
     pub async fn run(self) -> Result<()> {
         let mut client = spin_oci::Client::new(self.insecure, None).await?;
-        client.pull(&self.reference).await?;
 
+        let pb = ProgressBar::new_spinner();
```

Comment:
This pattern, which I think is used in both places, feels like it could be good to encapsulate e.g.

```
let spinner = SpinnySpinner::start("Pulling app from registry");
// Do the long-running work
match result {
  Case1 => spinner.success("Successfully pulled app from registry"),
  Case2 => spinner.error("Failed to pull app from registry"),
}
```

(The names are not great because they don't communicate stopping/finishing but hopefully gives a sense of what I mean... capturing the standard dots and tempo and styles into a simplified API.)